testsuite/MDAnalysisTests/analysis/test_lineardensity.py

```diff
@@ -25,15 +25,70 @@
 
 from MDAnalysisTests.datafiles import waterPSF, waterDCD
 from MDAnalysis.analysis.lineardensity import LinearDensity
-from numpy.testing import assert_almost_equal
+from numpy.testing import assert_allclose
 
 
-def test_serial():
+def test_grouping_atoms():
```

Comment:
You may simplify duplicated code with `pytest.mark.parametrize`. Look for examples in the testsuite, there are plenty of them. 

Additionally, there are no tests for the actual calculated density. I would at least add a check for the mean of `results.x.pos` and `results.x.char`

Reply:
You are right. The current values are just magic numbers. These will probably fail when we do something with the code. Depending on how much effort you want to invest I propose the following:

Generate a trajectory based on random positions of particles in a cubic box. Draw the positions from a uniform distribution. Run the analysis and compare the densities. You could do three kinds of system: uncharged particles, charged particles and maybe charged dimers. This is probably beyond the scope of this PR but would be a valuable addition!

Reply:
One reason why I stuck with binsize=5 with my new tests is that the original test, which also used this value, had expected values for `results.x.pos`. I thought that this would at least be partial validation for the restored functionality. But independently determined expected values are definitely still needed here. 

Thanks for the suggestion! For now, I need to focus on my GSoC application/proposal, but I would be interested in working on LinearDensity further. I originally started looking into this module because it looked like a good candidate for more tests on codecov, and that's when I encountered the bug fixed in this PR. It looks like between adding tests for the parallel analysis code, getting expected values for the tests, and issues #2507 and #2508, there is still quite a bit to be done here, and I'd be happy to look into it more after more work on my application. For now, I can open an issue describing the need for more test data, and this PR could get merged?

Reply:
Yes take your time but opening an issue is a good idea!

Reply:
I have now opened issue #3603 for improved tests.

Reply:
Just a quick comment: "ground truth" tests are excellent when you can get them. However, "regression tests" are the next best thing: they will check that future changes do not change what was once considered working. For this PR I would be totally happy with regression tests, especially as the original code also used them.

Thanks for raising the issue, that's a good approach to doing incremental software development: do small working units and keep track of what needs to be done through issues.